### PR TITLE
Add Jacoco Code Coverage support

### PR DESCRIPTION
**Pull request by anshul98ks123**
### Issue(s)

[[BE] Apps Team Testing & Secure CI Update - V1 (Manual)](https://startree.atlassian.net/browse/AI-18)

### Description

This PR adds basic jacoco code coverage local setup

### How to test
1. `./mvnw clean verify`
2. Check aggregate coverage report in 
```jacoco-code-coverage/target/site/jacoco-aggregate/index.html```

Screenshot 2024-12-16 at 8 57 03 PM

Also verified that maven release doesn't break by a local run
```
./mvnw clean -B -DskipTests -Darguments=-DskipTests release:clean initialize release:prepare -U
```
Output:
[thirdeye-release-logs.txt](https://github.com/user-attachments/files/18151998/thirdeye-release-logs.txt)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -70,6 +71,8 @@
     <hsqldb.version>2.7.1</hsqldb.version>
     <!-- should match dropwizard version - except for security issues -->
     <jackson.version>2.18.1</jackson.version>
+    <!-- Skip Jacoco by default -->
+    <jacoco.skip>true</jacoco.skip>
```
Comment (cyrilou242) [inline pom.xml:75]: what's the test time difference between jacoco enabled and disabled ? 
if it's not too much maybe we could have it run in all PRs

Comment (anshul98ks123) [reply pom.xml:75]: just 3 second difference
lemme remove the flag

**Comment by cyrilou242** [comment]
> Also added a profile skip-integration-tests to speedup jacoco report generation, as integration tests module are not needed while computing code coverage and -DskipTests is not useful since it skips all tests

I'm a bit confused by this. Best case jacoco should be able to measure codepaths hit by the integration tests.

**Comment by anshul98ks123** [comment]
> > Also added a profile skip-integration-tests to speedup jacoco report generation, as integration tests module are not needed while computing code coverage and -DskipTests is not useful since it skips all tests
> 
> I'm a bit confused by this. Best case jacoco should be able to measure codepaths hit by the integration tests.

you're right
i had to exclude integration tests path in dm code because it was taking like hours of time, but forgot the basic point that integration tests do test codepaths of other modules and contribute to the code coverage score

reverting that commit

**Comment by anshul98ks123** [comment]
done

**Review by cyrilou242 — APPROVED** [review]
LGTM
:shipit:

**Pull request merged**